src/components/input/input.scss

```diff
@@ -0,0 +1,153 @@
+@import 'default-theme';
+@import 'mixins';
+@import 'variables';
+
+
+//
+$md-input-label-color: md-color($md-foreground, hint-text);
+$md-input-floating-label-color: md-color($md-primary);
+$md-input-required-label-color: md-color($md-accent);
+$md-input-underline-color: md-color($md-foreground, hint-text);
+$md-input-underline-color-accent: md-color($md-accent);
+$md-input-underline-disabled-color: md-color($md-foreground, hint-text);
+$md-input-underline-focused-color: md-color($md-primary);
+
+
+:host {
+  display: inline-block;
+  position: relative;
+  font-family: $md-font-family;
+
+  .md-wrapper {
+    margin: 16px 0;
+  }
+  .md-flex {
+    display: inline-table;
+    flex-flow: column;
+    vertical-align: bottom;
+    width: 100%;
+  }
+  .md-flex > * {
+    display: table-cell;
+  }
+
+  input {
+    // Font needs to be inherited, because by default <input> has a system font.
+    font: inherit;
+
+    // By default, <input> has a padding, border, outline and a default width.
+    border: none;
+    outline: none;
+    padding: 0;
+    width: 100%;
+
+    &.md-end {
+      text-align: right;
+    }
+  }
+
+  label {
+    position: absolute;
+    visibility: hidden;
+    font-size: 100%;
+    pointer-events: none;
+    color: $md-input-label-color;
+    z-index: 1;
+
+    width: 100%;
+
+    transform: translateY(0);
+    transform-origin: bottom left;
+    transition: transform $swift-ease-out-duration $swift-ease-out-timing-function,
+                scale $swift-ease-out-duration $swift-ease-out-timing-function,
+                color $swift-ease-out-duration $swift-ease-out-timing-function;
+
+    &.md-empty {
+      visibility: visible;
+      cursor: text;
+    }
+
+    &.md-float:not(.md-empty), &.md-float.md-focused {
+      visibility: visible;
+      padding-bottom: 5px;
+      transform: translateY(-100%) scale(0.75);
+
+      span {
+        color: $md-input-required-label-color;
+      }
+    }
+
+    &.md-focused {
+      color: $md-input-floating-label-color;
+
+      &.md-accent {
+        color: $md-input-underline-color-accent;
+      }
+    }
+  }
+
+  .md-underline {
+    position: absolute;
+    height: 1px;
+    width: 100%;
+    margin-top: 4px;
+    border-top: 1px solid $md-input-underline-color;
+
+    &.md-disabled {
+      border-top: 0;
+      background-image: linear-gradient(to right, rgba(0,0,0,0.26) 0%, rgba(0,0,0,0.26) 33%, transparent 0%);
+      background-position: 0;
+      background-size: 4px 1px;
+      background-repeat: repeat-x;
+    }
+
+    > span {
```

Comment:
Definitely a class here; encountering just `> span` is very hard to map back to what's actually being styled in the template.